pkg/collectsub/datasource/csubsource/csubsource_test.go

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package csubsource
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/guacsec/guac/pkg/collectsub/client"
+	"github.com/guacsec/guac/pkg/collectsub/collectsub"
+	"github.com/guacsec/guac/pkg/collectsub/datasource"
+)
+
+func createSimpleCsubClient(ctx context.Context) (client.Client, error) {
+	c, err := client.NewMockClient()
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.AddCollectEntries(ctx, []*collectsub.CollectEntry{
+		{Type: collectsub.CollectDataType_DATATYPE_OCI, Value: "oci://abc"},
+		{Type: collectsub.CollectDataType_DATATYPE_OCI, Value: "oci://def"},
+		{Type: collectsub.CollectDataType_DATATYPE_GIT, Value: "git+https://github.com/guacsec/guac"},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func Test_FileSourceGetDataSources(t *testing.T) {
+	ctx := context.TODO()
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
+	defer cancel()
+
+	c, err := createSimpleCsubClient(ctx)
+	if err != nil {
+		t.Errorf("unable to initiliaze simple source client: %v", err)
+		return
+	}
+	defer c.Close()
+
+	cds, err := NewCsubDatasource(c, time.Second)
+	if err != nil {
+		t.Errorf("unable to create FileDataSources: %v", err)
+		return
+	}
+
+	ds, err := cds.GetDataSources(ctx)
+	if err != nil {
+		t.Errorf("unable to get DataSources: %v", err)
+		return
+	}
+
+	expected := &datasource.DataSources{
+		OciDataSources: []datasource.Source{
+			{Value: "oci://abc"},
+			{Value: "oci://def"},
+		},
+		GitDataSources: []datasource.Source{
+			{Value: "git+https://github.com/guacsec/guac"},
+		},
+	}
+
+	if !reflect.DeepEqual(ds, expected) {
+		t.Errorf("unexpected datasource output: expect %v, got %v", expected, ds)
+	}
+}
+
+func Test_FileSourceDataSourcesUpdate(t *testing.T) {
+	ctx := context.TODO()
+
+	c, err := createSimpleCsubClient(ctx)
+	if err != nil {
+		t.Errorf("unable to initiliaze simple source client: %v", err)
+		return
+	}
+	defer c.Close()
+
+	cds, err := NewCsubDatasource(c, time.Second)
+	if err != nil {
+		t.Errorf("unable to create FileDataSources: %v", err)
+		return
+	}
+
+	upChan, err := cds.DataSourcesUpdate(ctx)
+	if err != nil {
+		t.Errorf("unable to get DataSourcesUpdate: %v", err)
+	}
+
+	ds, err := cds.GetDataSources(ctx)
```

Comment:
is there a reason why `GetDataSources` gets called twice and compared in this test?

Reply:
yup - it gets updated and it checks to see if it got the updated content.